modules/extensions/src/views/lite/components/debugger/views/State.tsx

```diff
@@ -0,0 +1,82 @@
+import { Button } from '@blueprintjs/core'
+import * as sdk from 'botpress/sdk'
+import cx from 'classnames'
+import _ from 'lodash'
+import React, { FC, Fragment, useEffect, useState } from 'react'
+
+import lang from '../../../../lang'
+import { Collapsible } from '../components/Collapsible'
+import style from '../style.scss'
+
+import { Inspector } from './Inspector'
+
+interface Props {
+  state: sdk.IO.EventState
+  prevState: sdk.IO.EventState
+  isExpanded: (key: string) => boolean
+  toggleExpand: (section: string, expanded: boolean) => void
+}
+
+const STATE_JSON = 'json::state'
+const STATE_PANEL = 'panel::state'
+
+const State: FC<Props> = ({ state, prevState, isExpanded, toggleExpand }) => {
+  const [viewJSON, setViewJSON] = useState(isExpanded(STATE_JSON))
+
+  useEffect(() => {
+    setViewJSON(isExpanded(STATE_JSON))
+  }, [isExpanded(STATE_JSON)])
+
+  const toggleView = () => {
+    const newValue = !viewJSON
+    toggleExpand(STATE_JSON, newValue)
+    setViewJSON(newValue)
+  }
+
+  const renderContent = () => {
+    if (viewJSON) {
+      return <Inspector data={state} />
+    }
+
+    const variables = state.session.workflows[state.session.currentWorkflow]?.variables || {}
+    const prevVariables = prevState?.session.workflows[prevState.session.currentWorkflow]?.variables || {}
```

Comment:
Technically you want to compare the variables of the current workflow with those of the previous event, so you should use `state.session.currentWorkflow` in both requests. If you change workflow between the events it will display weird stuff if there is a similar name in differnet workflows